pkg/cmd/install.go

```diff
@@ -280,7 +280,8 @@ func (o *installCmdOptions) install(_ *cobra.Command, _ []string) error {
 			}
 		}
 
-		if !o.KanikoBuildCache {
+		kanikoBuildCacheFlag := cobraCmd.Flags().Lookup("kaniko-build-cache")
+		if kanikoBuildCacheFlag.Changed {
```

Comment:
why is this needed ? the KanikoBuildCache should be set already by the pre run task

Reply:
With #1121, the default is now `false`. With falsy default, we have the possibility to set the option even if it's not explicitly provided. Still, it seems more user-friendly not to fill the `Spec` struct with default values. Boolean is a special case as _zero_ value is a valid value.

Reply:
So we can just flip the if the were previously in place

Reply:
Yes, unless we find it more user-friendly not to fill the Spec struct with default values.

Reply:
but if you flip the switch, the default won't be set, isn't it ?

Reply:
Yes, but if the user explicitly set the option to `false`, it won't be set either, which doesn't really makes sense.

Reply:
ah ok, understood. 
so let merge it, problem is that this wont' work with the env + file thing.

I'll think about a solution later on